src/psm/src/node.cpp

```diff
@@ -153,6 +153,20 @@ bool Node::hasInstances() const
   return !connected_instances_.empty();
 }
 
+void Node::setEnclosure(NodeEnclosure encl)
+{
+  has_enclosure_ = true;
+  encl_ = encl;
+}
+NodeEnclosure Node::getEnclosure() const
+{
+  if(has_enclosure_){
+    return encl_;
+  } else {
+    return NodeEnclosure{0,0,0,0};
```

Comment:
This is the default value for encl_ so why do we need has_enclosure_?  From the outside nobody can tell the difference between no enclosure and this value.